app/ante/max_tx_size_test.go

```diff
@@ -0,0 +1,71 @@
+package ante_test
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v3/app/ante"
+	v2 "github.com/celestiaorg/celestia-app/v3/pkg/appconsts/v2"
+	v3 "github.com/celestiaorg/celestia-app/v3/pkg/appconsts/v3"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	version "github.com/tendermint/tendermint/proto/tendermint/version"
+)
+
+func TestMaxTxSizeDecorator(t *testing.T) {
+	decorator := ante.NewMaxTxSizeDecorator()
+	anteHandler := sdk.ChainAnteDecorators(decorator)
+
+	testCases := []struct {
+		name        string
+		txSize      int
+		expectError bool
+		appVersion  uint64
+	}{
```

Comment:
optional, but useful 

we can add a bool to determine if this is checkTx here, then add that bool to the test sdk.Context as will with `sdkCtx.WithIsCheckTx(true)`

Reply:
both addressed https://github.com/celestiaorg/celestia-app/pull/3909/commits/54b2b4d1ae3874579b1617aee8da258be4bd7f4c